osu.Game/Screens/Play/MenuOverlay.cs

```diff
@@ -182,6 +184,42 @@ private void load(OsuColour colours)
             Retries = 0;
         }
 
+        protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)
+        {
+            foreach (DialogButton button in Buttons)
+                button.IsMouseSelected = false;
+
+            if (!args.Repeat && args.Key == Key.Up)
+            {
+                selectedButton--;
+
+                if (selectedButton < 0)
+                    selectedButton = Buttons.Count - 1;
+
+                Buttons[selectedButton].IsKeyboardSelected = true;

```

Comment:
This needs to be setting `IsKeyboardSelected = false` on the last keyboard-selected button.